### ISSUE TITLE
manualFetchRates and multiple providers not working as intended  

### ISSUE THREAD

**Issue by zollf** [verbb/postie#96]
### Describe the bug

When `manualFetchRates` is enabled and attempting to fetch shipping rates from multiple providers, the first provider is the only one that fetches shipping rates. I believe the issue here is from:
https://github.com/verbb/postie/blob/e28a422993c26ae33ccd693218a5292200f50177/src/base/Provider.php#L415-L422
This code is being called per provider, so the first provider will fetch shipping rates and will remove the session key. The subsequent providers will skip since `postieManualFetchRates` is no longer true. 
https://github.com/verbb/postie/blob/e28a422993c26ae33ccd693218a5292200f50177/src/base/Provider.php#L440-L447

### Steps to reproduce

1. Set `manualFetchRates` to true in settings
2. Create cart, add products and shipping address, enough to be able to get a shipping estimate from multiple providers
3. Send `POST` to update cart route with `fetchRatesPostValue` to trigger manually fetch shipping. 
4. Only one results return when many should

If a more verbose replication is needed please let me know.

### Craft CMS version

Craft Pro 3.7.58

### Plugin version

2.4.20

### Multi-site?

No

### Additional context

PHP Version: 7.4.30
OS version: Linux 5.10.124-linuxkit
Craft Commerce: 3.4.10.1

**Comment by engram-design** [comment]
Fixed for the next release. To get this early, change your `verbb/postie` requirement in `composer.json` to:

```
"require": {
  "verbb/postie": "dev-craft-3 as 2.4.20",
  "...": "..."
}
```

Then run `composer update`.

**Comment by zollf** [comment]
@engram-design thanks! 🥳

**Comment by engram-design** [comment]
Fixed in [2.4.21](https://github.com/verbb/postie/releases/tag/2.4.21)
